metrics/web/test/auth/domain/usercases/create_user_profile_usecase_test.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:metrics/auth/domain/entities/theme_type.dart';
+import 'package:metrics/auth/domain/usecases/create_user_profile_usecase.dart';
+import 'package:metrics/auth/domain/usecases/parameters/user_profile_param.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../../test_utils/user_repository_mock.dart';
+
+void main() {
+  group("CreateUserProfileUseCase", () {
+    final repository = UserRepositoryMock();
+    const id = 'id';
+    const selectedTheme = ThemeType.dark;
+    final userProfileParam = UserProfileParam(
+      id: id,
+      selectedTheme: selectedTheme,
+    );
+
+    tearDown(() {
+      reset(repository);
+    });
+
+    test("throws an ArgumentError if the given repository is null", () {
+      expect(
+        () => CreateUserProfileUseCase(null),
+        throwsArgumentError,
+      );
+    });
+
+    test("delegates call to the given repository", () {
+      final createProfile = CreateUserProfileUseCase(repository);
+
+      createProfile(userProfileParam);
+
+      verify(repository.createUserProfile(id, selectedTheme)).called(equals(1));
+    });
+
+    test("throws if the given repository throws during creating profile", () {
+      const errorMessage = 'error message';
+
+      when(repository.createUserProfile(any, any)).thenThrow(errorMessage);
+
+      final createProfile = CreateUserProfileUseCase(repository);
+
+      expect(
+        () => createProfile(userProfileParam),
+        throwsA(equals(errorMessage)),
+      );
```

Comment:
```suggestion
      final result = createProfile(userProfileParam);
      
      expect(
        result,
        throwsA(equals(errorMessage)),
      );
```
I'm not sure but it seems like we can use `Future`s directly without wrapping it in callback.